### PR TITLE
Update deployment flow tests to enzyme

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
[Update deployment flow tests to use Enzyme](https://trello.com/c/JJjcpwPi/404-update-deployment-flow-tests-to-use-enzyme)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8574/

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 1]
```diff
@@ -166,6 +166,9 @@ class DeploymentCredentialAdd extends React.Component {
     if (!info || !info.forms) {
       return;
     }
+    if (!info.forms[this.state.authType]) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: Was this a bug?

Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: It appeared to be, yes. I don't think we ever hit the bug due to the props we had, but I suspect we would have hit it down the track depending on the cloud configs.

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**